specs/release.md

```diff
@@ -0,0 +1,319 @@
+
+Direct commits to `main` are PROHIBITED except for emergency hotfixes when the pull request process would create unacceptable delay. Emergency commits MUST be documented with justification in the commit message.
+
+### Feature Branches
+
+All development work MUST occur on feature branches created from `main`. Feature branches SHOULD follow the naming convention `feature/{description}` or `fix/{description}` to indicate intent. Branch names MUST use kebab-case and SHOULD be concise but descriptive (e.g., `feature/oauth-pkce-support`, `fix/session-timeout-bug`).
+
+Feature branches MUST be short-lived, typically merged within 1-3 days of creation. Long-running feature branches increase merge conflicts and defer integration issues. For large features requiring longer development, use feature flags to merge incomplete work while keeping the functionality disabled in production.
+
+Feature branches SHOULD be deleted immediately after merge to reduce repository clutter and prevent accidental continued development on stale branches.
+
+### Hotfix Workflow
+
+Critical production issues requiring immediate deployment MAY bypass the standard pull request review process. Hotfixes MUST:
+
+1. Create a branch from `main` (not from a feature branch)
+2. Include only the minimal changes required to resolve the critical issue
+3. Add or update tests to prevent regression
+4. Deploy to development environment for validation before production
+5. Create a pull request for post-deployment review and documentation
+
+Hotfixes MUST be deployed to production within 1 hour of identification for critical failures (authentication broken, data loss risk, security vulnerability). Non-critical issues SHOULD follow the standard release process.
+
+## Continuous Integration Pipeline
+
+Every push to any branch and every pull request MUST trigger automated CI checks that validate code quality, correctness, and test coverage.
+
+### CI Workflow Stages
+
+The CI pipeline MUST execute the following stages in order, failing fast at the first error:
+
+**1. Type Checking:** TypeScript compilation MUST succeed with no type errors. The project uses strict mode and all type checking rules MUST be enforced.
+
+**2. Unit Tests:** All unit tests MUST pass with deterministic results. Flaky tests MUST be fixed or quarantined. The test suite MUST complete in under 60 seconds to provide rapid feedback.
+
+**3. E2E Tests:** End-to-end integration tests MUST pass, validating the full OAuth 2.1 + PKCE + MCP protocol flow. E2E tests MUST complete in under 30 seconds and MUST NOT require manual intervention or browser interaction.
+
+**4. Coverage Validation:** Code coverage MUST meet minimum thresholds (95% statement coverage, 95% function coverage). Coverage regressions MUST fail the build.
+
+### CI Execution Environment
+
+CI jobs MUST run on GitHub Actions using Ubuntu latest runners. The environment MUST use:
+
+- Node.js 20 (LTS)
```

Comment:
Can we just mandate here that we use the node version specified for mise and the pnpm version specified in package.json? We just want to mandate they're consistent rather than a specific version.